src/btree/bt_debug.c

```diff
@@ -801,44 +775,66 @@ __wt_debug_disk(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, const char
     WT_DBG *ds, _ds;
     WT_DECL_RET;
     uint32_t flags;
+    bool is_first;
 
     ds = &_ds;
+    is_first = true;
 
     WT_ASSERT(session, !(dump_all_data && dump_key_data));
     flags = dump_all_data ? WT_DEBUG_UNREDACT_ALL : 0;
     flags |= dump_key_data ? WT_DEBUG_UNREDACT_KEYS : 0;
     WT_RET(__debug_config(session, ds, ofile, flags));
 
-    WT_ERR(ds->f(ds, "%s page", __wt_page_type_string(dsk->type)));
+    WT_ERR(ds->f(ds, "- %s page\n\t> ", __wt_page_type_string(dsk->type)));
     switch (dsk->type) {
     case WT_PAGE_BLOCK_MANAGER:
         break;
     case WT_PAGE_COL_FIX:
     case WT_PAGE_COL_INT:
     case WT_PAGE_COL_VAR:
-        WT_ERR(ds->f(ds, ", recno %" PRIu64, dsk->recno));
+        WT_ERR(ds->f(ds, "recno: %" PRIu64 " | ", dsk->recno));
     /* FALLTHROUGH */
     case WT_PAGE_ROW_INT:
     case WT_PAGE_ROW_LEAF:
-        WT_ERR(ds->f(ds, ", entries %" PRIu32, dsk->u.entries));
+        WT_ERR(ds->f(ds, "entries: %" PRIu32 " | ", dsk->u.entries));
         break;
     case WT_PAGE_OVFL:
-        WT_ERR(ds->f(ds, ", datalen %" PRIu32, dsk->u.datalen));
+        WT_ERR(ds->f(ds, "datalen: %" PRIu32 " | ", dsk->u.datalen));
         break;
     default:
         WT_ERR(__wt_illegal_value(session, dsk->type));
     }
 
-    if (F_ISSET(dsk, WT_PAGE_COMPRESSED))
-        WT_ERR(ds->f(ds, ", compressed"));
-    if (F_ISSET(dsk, WT_PAGE_ENCRYPTED))
-        WT_ERR(ds->f(ds, ", encrypted"));
-    if (F_ISSET(dsk, WT_PAGE_EMPTY_V_ALL))
-        WT_ERR(ds->f(ds, ", empty-all"));
-    if (F_ISSET(dsk, WT_PAGE_EMPTY_V_NONE))
-        WT_ERR(ds->f(ds, ", empty-none"));
+    if (dsk->flags != 0) {
+        WT_ERR(ds->f(ds, "dsk_flags: ["));
+
+        if (F_ISSET(dsk, WT_PAGE_COMPRESSED)) {
+            WT_ERR(ds->f(ds, "compressed"));
+            is_first = false;
```

Comment:
apologies if i've missed something but how are u handling trailing commas in this chunk of code?

Reply:
I'm using the `is_first` flag keeps track of the first variable in the `dsk_flags` list and I then use ternary operators in the print statements to decide if a comma is necessary. I.e. 
```
WT_ERR(ds->f(ds, "%sencrypted", is_first ? "" : ", "));
```